pet-api/pet-api/Infrastructure/DAL/Repositories/ProductRepository.cs

```diff
@@ -13,24 +13,35 @@ public class ProductRepository : IRepository<Product, ProductModel>
     {
         private readonly SqlConnection _connection;
         private readonly DbTransaction _transaction;
+        private readonly IConfiguration _configuration;
 
-        public ProductRepository(SqlConnection connection, DbTransaction transaction)
+        public ProductRepository(SqlConnection connection, DbTransaction transaction, IConfiguration configuration)
```

Comment:
Do not pass IConfiguration to the Repository. The repository is responsible for the DAL layer, you are using the IConfiguration for mapping/building URLs reasons. I think you can store just an image name + extension and build a URL in some mapping service. It would be better to avoid this dependency.

Reply:
I will add the mapper, that will be used in the ProductController to build ProductDTO and correct URL before data-transfer